### PR TITLE
Avoid uninformative warnings

### LINKED ISSUE #174: Investigate various warnings
Casanovo issues a bunch of warning messages when I run it on a linux system.  We should investigate and try to eliminate them.  Here are the ones I am seeing:

```
2023-04-17 15:12:34,632 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/noble/miniconda3/envs/casanovo/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/accelerator_connector.py:589: LightningDeprecationWarning: The Trainer argument `auto_select_gpus` has been deprecated in v1.9.0 and will be removed in v2.0.0. Please use the function `pytorch_lightning.accelerators.find_usable_cuda_devices` instead.
  rank_zero_deprecation(

2023-04-17 15:12:52,558 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/noble/miniconda3/envs/casanovo/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, predict_dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 48 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(

2023-04-17 15:13:42,405 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/noble/miniconda3/envs/casanovo/lib/python3.10/site-packages/torch/nn/modules/transformer.py:287: UserWarning: The PyTorch API of nested tensors is in prototype stage and will change in the near future. (Triggered internally at ../aten/src/ATen/NestedTensorImpl.cpp:177.)
  output = torch._nested_tensor_from_mask(output, src_key_padding_mask.logical_not(), mask_check=False)

2023-04-17 15:13:42,780 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/noble/miniconda3/envs/casanovo/lib/python3.10/site-packages/torch/nn/functional.py:4999: UserWarning: Support for mismatched key_padding_mask and attn_mask is deprecated. Use same type for both instead.
  warnings.warn(

2023-04-17 15:13:42,838 WARNING [py.warnings/MainProcess] warnings._showwarnmsg : /net/noble/vol1/home/noble/miniconda3/envs/casanovo/lib/python3.10/site-packages/torch/nn/modules/activation.py:1144: UserWarning: Converting mask without torch.bool dtype to bool; this will negatively affect performance. Prefer to use a boolean mask directly. (Triggered internally at ../aten/src/ATen/native/transformers/attention.cpp:150.)
  return torch._native_multi_head_attention(
```

### PR DESCRIPTION
**Pull request by bittremieux**
Fixes #174.

### CODE REVIEW

**Comment by bittremieux** [comment]
To keep a record:
- Message `UserWarning: Converting mask without torch.bool dtype to bool; this will negatively affect performance. Prefer to use a boolean mask directly. (Triggered internally at ../aten/src/ATen/native/transformers/attention.cpp:150.)` doesn't make sense, because even when provided with boolean masks, `torch.nn.modules.activation.forward` converts the masks to the same dtype as the query, which is float32.

**Review by VarunAnanth2003 — APPROVED** [review]
Looks good, see previous comment for testing details (works as intended)
